src/net/parsers/texttracks/webvtt.js

```diff
@@ -0,0 +1,311 @@
+/** 
+* Parse WebVTT from text. Returns an array with:
+* - start : start of current cue, in seconds
+* - end : end of current cue, in seconds
+* - content : HTML formatted cue.
+*
+* Global style is parsed and applied to div element. 
+* Specific style is parsed and applied to class element. 
+*
+* @param {string} text
+* @return {Array<Object>}
+*/
+export default function parseWebVTT(text) {
+  const newLineChar = /\n|\s{2,}/g;
+  const linified = text.split(newLineChar);
+  const cuesArray = [];
+  const styleElements = [];
+
+  if (!linified[0].match(/^WEBVTT.*/)) {
+    console.log("Can't parse WebVTT: Non valid file.");
+    return;
+  }
+
+  let i = 0;
+
+  // Find every style element into WebVTT, and collect style 
+  // associated with class / tags
+  for (i = 1; i<linified.length; i++) {
+    if (!(linified[i].match(/^\r/) || linified[i].length === 0)) {
+      if(findElementType(linified[i]) === "style"){
+        parseStyle(i+1, linified, styleElements);        
+      }
+      while (linified[i] 
+        && !(linified[i].match(/^\r/) || linified[i].length === 0)) {
+        i++;
+      }
+    }
+  }
+
+  // Parse cues, format and apply style.
+  for (i = 1; i < linified.length; i++) {
+    if (!(linified[i].match(/^\r/) || linified[i].length === 0)) {
+      if (findElementType(linified[i]) === "cue") {
+        cuesArray.push(parseCue(i, linified, styleElements));
+      }
+      while (linified[i] 
+        && !(linified[i].match(/^\r/) || linified[i].length === 0)) {
+        i++;
+      }
+    }
+  }
+
+  return cuesArray;
+}
+
+/**
+ * Finds from supposed first line of element 
+ * (WebVTT is line based format) the type of it. It can be:
+ * - note: A comment
+ * - style: A style indication, about all cues, or specific cues.
+ * - cue: The subtitle content.
+ * - region: Informations about wrapping area of subtitle
+ * @param {string} text 
+ */
+function findElementType(text) {
+  if (text.match(/^NOTE.*?/g)) {
+    return "note";
+  }
+  else if (text.match(/^STYLE.*?/g)) {
+    return "style";
+  }
+  else if (text.match(/^REGION.*?/g)) {
+    return "style";
+  }
+  else if(text.length !== 0){
+    return "cue";
+  }
+}
+
+/**
+ * 
+ * Parse style element from WebVTT.
+ * @param {number} index 
+ * @param {Array<string>} linified 
+ * @param {Array<Object>} styleElements 
+ */
+function parseStyle(index, linified, styleElements) {
+
+  const className = [];
+  const cueClassLine = linified[index].match(/cue\(\.{0,1}(.*?)\)/);
+  className.push(cueClassLine ? cueClassLine[1] : "cue");
+
+  index++;
+
+  while(linified[index].match(/cue\(\.{0,1}(.*?)\)/)) {
+    const cueClassLine = linified[index].match(/cue\(\.{0,1}(.*?)\)/);
+    className.push(cueClassLine ? cueClassLine[1] : "cue");
+    index++;
+  }
+
+  let styleContent = "";
+
+  while (!(linified[index].match(/}/))) {
+    styleContent += linified[index];
+    index++;
+  }
+
+  className.forEach(c => {
+    styleElements.push({
```

Comment:
So here, styleElements is only for constructing it by side effects?

Wouldn't it be more logical than parseStyle returns the style objects that it parsed and it's the caller's role to add it wherever it wants?